prov/util/src/rxm_av.c

```diff
@@ -98,23 +98,29 @@ static void rxm_free_peer(struct util_peer_addr *peer)
 	ofi_ibuf_free(peer);
 }
 
-struct util_peer_addr *
-util_get_peer(struct rxm_av *av, const void *addr)
+
+int util_get_peer(struct rxm_av *av, const void *addr, struct util_peer_addr **peer,
+		  uint64_t flags)
 {
-	struct util_peer_addr *peer;
 	struct ofi_rbnode *node;
+	int ret;
 
 	ofi_mutex_lock(&av->util_av.lock);
 	node = ofi_rbmap_find(&av->addr_map, (void *) addr);
 	if (node) {
-		peer = node->data;
-		peer->refcnt++;
+		*peer = node->data;
+		(*peer)->refcnt++;
```

Comment:
Set ret = FI_SUCCESS here to avoid an extra branch on return.

Reply:
Something is off in the flag settings.  A peer might be inserted in response to a connection request (xnet_process_connreq).  In that case, peer->av_flags = 0.  When an address is inserted with FI_FIREWALL_ADDR, the peer will be found, but peer->av_flags won't be adjusted.

I think you need ```(*peer)->av_flags |= flags;``` here.